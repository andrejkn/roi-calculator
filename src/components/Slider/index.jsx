import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const SliderContainer = styled.div`
  border-radius: 0px;
  border-color: #888;
  margin: 10px 0px;
`;

const LabelContainer = styled.div`
  text-align: center;
`;

const SliderElement = styled.input`
  width: 100%;
`;

export default function Slider({
  value,
  label,
  minValue,
  maxValue,
  step,
  onChange,
}) {
  return (
    <SliderContainer>
      <LabelContainer>
        {label}
      </LabelContainer>
      <SliderElement
        type="range"
        min={minValue}
        max={maxValue}
        step={step}
        value={value}
        onChange={onChange}
      />
    </SliderContainer>
  );
}

Slider.defaultProps = {
  step: 1,
  label: '',
};

Slider.propTypes = {
  onChange: PropTypes.func.isRequired,
  value: PropTypes.number.isRequired,
  minValue: PropTypes.number.isRequired,
  maxValue: PropTypes.number.isRequired,
  step: PropTypes.number,
  label: PropTypes.string,
};
