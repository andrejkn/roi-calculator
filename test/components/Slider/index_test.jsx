import React from 'react';
import renderer from 'react-test-renderer';

import Slider from '../../../src/components/Slider';

let component = null;
let testProps = null;

describe('Slider component test', () => {
  beforeEach(() => {
    testProps = {
      onChange: jest.fn(),
      value: 1000,
      minValue: 10,
      maxValue: 500,
    };

    component = renderer.create(
      <Slider
        {...testProps}
      />,
    );
  });

  it('should match snapshot', () => {
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should trigger onChange', () => {
    expect(testProps.onChange).toHaveBeenCalledTimes(0);
    renderer.act(component.root.props.onChange);
    expect(testProps.onChange).toHaveBeenCalledTimes(1);
  });
});
