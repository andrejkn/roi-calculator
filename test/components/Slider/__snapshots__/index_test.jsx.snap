// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider component test should match snapshot 1`] = `
<div
  className="sc-bdVaJa kuQLoz"
>
  <div
    className="sc-bwzfXH gJIwTT"
  >
    
  </div>
  <input
    className="sc-htpNat daVAfG"
    max={500}
    min={10}
    onChange={[MockFunction]}
    step={1}
    type="range"
    value={1000}
  />
</div>
`;
